python/cuml/_thirdparty/sklearn/preprocessing/_data.py

```diff
@@ -1936,6 +1974,72 @@ def transform(self, X, copy=None):
         return binarize(X, threshold=self.threshold, copy=copy)
 
 
+@api_return_generic(get_output_type=True)
+def add_dummy_feature(X, value=1.0):
+    """Augment dataset with an additional dummy feature.
+
+    This is useful for fitting an intercept term with implementations which
+    cannot otherwise fit it directly.
+
+    Parameters
+    ----------
+    X : {array-like, sparse matrix}, shape [n_samples, n_features]
+        Data.
+
+    value : float
+        Value to use for the dummy feature.
+
+    Returns
+    -------
+
+    X : {array, sparse matrix}, shape [n_samples, n_features + 1]
+        Same data with dummy feature added as first column.
+
+    Examples
+    --------
+
+    >>> from cuml.preprocessing import add_dummy_feature
+    >>> add_dummy_feature([[0, 1], [1, 0]])
+    array([[1., 0., 1.],
+           [1., 1., 0.]])
+    """
+    X = check_array(X, accept_sparse=['csc', 'csr', 'coo'], dtype=FLOAT_DTYPES)
+    n_samples, n_features = X.shape
+    shape = (n_samples, n_features + 1)
+    if sparse.issparse(X):
+        if sparse.isspmatrix_coo(X):
+            # Shift columns to the right.
+            col = X.col + 1
+            # Column indices of dummy feature are 0 everywhere.
+            col = np.concatenate((np.zeros(n_samples), col))
+            # Row indices of dummy feature are 0, ..., n_samples-1.
+            row = np.concatenate((np.arange(n_samples), X.row))
+            # Prepend the dummy feature n_samples times.
+            data = np.concatenate((np.full(n_samples, value), X.data))
+            X = sparse.coo_matrix((data, (row, col)), shape)
+            return X
+        elif sparse.isspmatrix_csc(X):
```

Comment:
Are there tests for COO and CSC?
Also, is there CSR support? Seems like that is probably an interesting case?

Reply:
Sure, there is support for CSR. CSR matrices get converted to COO and the function is ran again. CSR and CSC are tested. I'll add COO testing.